File: 04 자바기초/source/day06/src/exam01/Ex05.java
package exam01;

public class Ex05 {
    public static void main(String[] args) {
        int num1 = 10;
        int num2 = 2;

        try {
            int result = num1 / num2; // ArithmethicException

            String str = null;
            str = str.toUpperCase(); // 소문자를 대문자로 NullPointerException
            System.out.println(str);
        } catch (RuntimeException e) {
            String message = e.getMessage();
            System.out.println(message);

            if(e instanceof NullPointerException){
                System.out.println("NullPointerException 별도 처리");
            }
        }
    }
}

File: 03 js_class/js/ex05.js
// 즉시 실행 함수
(function(){
    console.log("즉시 실행");
})();

(function (a,b){
    const result= a + b;
    console.log(result);
})(10, 20);

const val = (function (a,b){
    const result= a + b;
    return result;
})(10, 20);

console.log(val);